src/SmoFacade/Server.cs

```diff
@@ -224,7 +224,29 @@ public Database Database(string dbName)
 
         policy.Execute(() => restore.SqlRestore(_server));
         restore.Devices.Remove(backupDeviceItem);
+
+        if(fileRelocation != null)
+          LogicalFileRename(databaseName, fileRelocation);
+      }
+    }
+
+    private void LogicalFileRename(string databaseName, Func<string, string> fileRelocation)
+    {
+      var db = Database(databaseName);
+      if (db.Restoring)
+        db.RestoreWithRecovery();
+
+      foreach (FileGroup fileGroup in (SmoCollectionBase)db._database.FileGroups)
+      {
+        DataFile[] array = new DataFile[fileGroup.Files.Count];

```

Comment:
Please rename these arrays to something more meaningful.
`dataFiles` and `logFiles` seems to work?